src/ai/backend/manager/models/image.py

```diff
@@ -673,14 +673,30 @@ def __repr__(self) -> str:
     @property
     def resources(self) -> Resources:
         custom_resources = self._resources or {}
-        label_resources = self.get_resources_from_labels()  # Set this as default
-
-        resources = dict(custom_resources)
-        for label_key, value in label_resources.items():
-            if label_key not in resources:
-                resources[label_key] = value
-
-        return ImageRow._resources.type._schema.check(resources)
+        label_resources = self.get_resources_from_labels()
+
+        merged_resources: dict[str, Any] = {}
+
+        for label_key in {*custom_resources.keys(), *label_resources.keys()}:
+            custom_spec = custom_resources.get(label_key, {})
+            label_spec = label_resources.get(label_key, {})
+
+            merged_spec = dict(custom_spec)
+            for label_spec_key, value in label_spec.items():
+                merged_spec.setdefault(label_spec_key, value)
+
+            # TODO: Consider other slot types
```

Comment:
In principle, we should consider other slots as well, but this PR is closer to a hotfix.
So, let's proceed with this PR for now, and work on improving the structure in a more appropriate direction in a follow-up PR.